### PR TITLE
conversions for Eigen::Isometry3d in ROS 2 - re-iteration

### PR DESCRIPTION
**Pull request by vmayoral**
Follows from https://github.com/ros2/geometry2/pull/93/commits

### CODE REVIEW

**Comment by vmayoral** [comment]
So I went ahead and manually cherry-picked all relevant changes from https://github.com/ros2/geometry2/pull/93 in here. 

The diff now is actually rather silly so it seems that most changes are already in the `ros2` branch for some reason.

**Comment by vmayoral** [comment]
All right, I can see most of these changes got already in https://github.com/ros2/geometry2/pull/90/commits. Maybe @dirk-thomas you can squash all of them into a single small commit with "minor adjustments to tf2_eigen" or should we simply discard these changes?

**Comment by clalancette** [comment]
> So I went ahead and manually cherry-picked all relevant changes from #93 in here.

Cool, thanks.

> 
> The diff now is actually rather silly so it seems that most changes are already in the `ros2` branch for some reason.

Right.  What's left here is actually formatting changes, and the removal of one method.  Do we need this PR at all anymore?

File: tf2_eigen/include/tf2_eigen/tf2_eigen.h [hunk 1]
```diff
@@ -341,22 +341,6 @@ geometry_msgs::msg::Pose toMsg(const Eigen::Isometry3d& in) {
   return msg;
 }
 
-/** \brief Convert a Eigen::Vector3d type to a geometry_msgs::msg::Vector3.
```
Comment (vmayoral) [inline tf2_eigen/include/tf2_eigen/tf2_eigen.h:344]: This shouldn't be removed. Needed by `collision_detection` submodule of moveit2.

**Comment by vmayoral** [comment]
Closing here, it modifies things that shouldn't be modified. Let me know @vmayoral or @dirk-thomas if you miss anything but I'd say we're good now for moveit2.

**Comment by vmayoral** [comment]
@ahcorde can you advise if https://github.com/ros2/geometry2/pull/113/commits/6dccebe0e8b4d9ade2ea77c566d5db134e9005d6 is still needed please?

**Comment by ahcorde** [comment]
Yes, otherwise

> The negative w values are completely valid quaternions. I'd rather not just normalize it here. This breaks the expectation that fromMsg(toMsg(...)) will return the same value.

**Comment by vmayoral** [comment]
Ok, can you make a PR against `ros2` then with only that last commit of yours or should I make it myself?

**Pull request closed without merging**